src/web/server/app/controllers/apiGroups.ts

```diff
@@ -0,0 +1,153 @@
+import { Request, Response } from 'express';
+
+import * as authServices from '../services/auth';
+import * as apiGroupsServices from '../services/apiGroups';
+import * as apiServices from '../services/api';
+
+/**
+ * Adds a user to a group
+ */
+export const addUserGroup = async (req: Request, res: Response) => {
+  try {
+    const userResult = await authServices.retrieveUser(req.body.username);
+    if (userResult === undefined || userResult.rows.length < 1) {
+      res.status(400).send('Unable to find user');
+    } else {
+      await apiGroupsServices.addUserGroup(
+        userResult.rows[0].user_id,
+        req.body.groupId
+      );
+      res.status(200).send('OK');
+    }
+  } catch (e) {
+    console.error(e);
+    res.status(500).send('Server error');
+  }
+};
+
+/**
+ * Sends a list of streamIds for a cameraId if the user owns the camera or is in its group
+ */
+export const retrieveStreamIdsGroups = async (req: Request, res: Response) => {
+  const isOwner = await apiServices.verifyUserCamera(
+    req.session!.userId,
+    req.params.cameraId
```

Comment:
Should check if `req.params.cameraId` exists

Reply:
yes